### PR TITLE
Run integration tests in Travis + other testing updates

### PR DESCRIPTION
**Pull request by shoekstra**
- Update test framework to be in line with other SC cookbooks.
- Aim to test all mentioned supported OSes

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -1,117 +1,120 @@
...
-- name: freebsd-10.3
-- name: amazon
-  driver:
-    box: mvbcoding/awslinux
-- name: opensuse-13.2
+  - name: amazonlinux
+    driver_config:
+      box: mvbcoding/awslinux
+  - name: centos-7.3
+  - name: debian-8.6
```
Comment (tas50) [inline .kitchen.yml]: We should get Debian 9 in here as well

File: .kitchen.yml [hunk 2]
```diff
@@ -1,117 +1,120 @@
...
+    run_list:
+    - recipe[apt::default]
+  - name: fedora-25
+  - name: freebsd-10.3
+  - name: opensuse-13.2
```
Comment (tas50) [inline .kitchen.yml]: 13.2 is EOL and should get replaced with opensuse-leap-42.2

File: .travis.yml [hunk 3]
```diff
@@ -1,28 +1,48 @@
-# Use Travis's cointainer based infrastructure
-sudo: false
+sudo: required
+dist: trusty
 
 addons:
   apt:
     sources:
-      - chef-current-precise
+      - chef-stable-trusty
```
Comment (tas50) [inline .travis.yml]: We can test on current again now that we don't delete the promoted build out of current when we take a build to stable.

File: .travis.yml [hunk 4]
```diff
@@ -1,28 +1,48 @@
...
+  - INSTANCE=event-ubuntu-1604
+
 before_script:
+  - sudo iptables -L DOCKER || ( echo "DOCKER iptables chain missing" ; sudo iptables -N DOCKER )
   - eval "$(/opt/chefdk/bin/chef shell-init bash)"
-  # We have to install chef-sugar for ChefSpec
-  - /opt/chefdk/embedded/bin/chef gem install chef-sugar
-
-script:
   - /opt/chefdk/embedded/bin/chef --version
```
Comment (tas50) [inline .travis.yml]: /opt/chefdk/embedded/bin/ should get removed everywhere in this config now

File: spec/support/shared_examples_for_apache_module.rb [hunk 5]
```diff
@@ -10,7 +10,6 @@
   module_name = a2module
   module_enable = true
   module_conf = a2conf
-  module_identifier = "#{module_name}_module"
```
Comment (tas50) [inline spec/support/shared_examples_for_apache_module.rb]: ?

Comment (shoekstra) [inline spec/support/shared_examples_for_apache_module.rb]: cookstyle was complaining about it being a useless assignment so I removed it.

Comment (tas50) [inline spec/support/shared_examples_for_apache_module.rb]: works for me then :)

**Comment by shoekstra** [comment]
LGTM!  Awesome work.

**Comment by damacus** [comment]
Apart from that one minor comment that looks great!

File: Berksfile [hunk 6]
```diff
@@ -3,11 +3,6 @@ source 'https://supermarket.chef.io'
 metadata
 
 group :integration do
-  cookbook 'apt'
   cookbook 'fqdn', git: 'https://github.com/drpebcak/fqdn-cookbook.git'
```
Comment (shoekstra) [inline Berksfile]: Looks like  this isn't being used and can be removed?

Comment (tas50) [inline Berksfile]: we include fqdn in the runlist in testing. It's probably something we could wrap into the setup recipe there though

**Pull request merged**